pilot/pkg/serviceregistry/serviceentry/conversion_test.go

```diff
@@ -649,101 +702,101 @@ func testConvertServiceBody(t *testing.T) {
 			externalSvc: httpNone,
 			services: []*model.Service{
 				makeService("*.google.com", "httpNone", constants.UnspecifiedIP,
-					map[string]int{"http-number": 80, "http2-number": 8080}, true, model.Passthrough),
+					map[string][]int{"http-number": {80, 0}, "http2-number": {8080, 0}}, true, model.Passthrough),
```

Comment:
Is the idea here that if 

`servicePort` == `targetPort`, `targetPort` will be 0?

Reply:
`targetPort == 0` means it's unspecified(in the API)

> Should we not just always set it to == `servicePort`?

If so we would not be able to tell if `targetPort` is explicitly set to `servicePort` or unspecified